src/ImageRunModal.jsx

```diff
@@ -0,0 +1,177 @@
+                        dialogErrorDetail: cockpit.format("$0: $1", ex.error, ex.parameters && ex.parameters.reason)
+                    });
+                });
+    }
+
+    onValueChanged(key, value) {
+        this.setState({ [key]: value });
+    }
+
+    render() {
+        const { image } = this.props;
+        const dialogValues = this.state;
+
+        let defaultBody = (
+            <div className='ct-form-layout'>
+                <label className='control-label' htmlFor='run-image-dialog-image'>
+                    {_("Image")}
+                </label>
+                <div id='run-image-dialog-image'> { image.repoTags ? image.repoTags[0] : "" } </div>
+
+                <label className='control-label' htmlFor='run-image-dialog-name'>
+                    {_("Name")}
+                </label>
+                <input id='run-image-dialog-name'
+                    type='text'
+                    placeholder={_("Container Name")}
+                    value={dialogValues.containerName}
+                    onChange={e => this.onValueChanged('containerName', e.target.value)}
+                    className='form-control' />
+
+                <label className='control-label' htmlFor='run-image-dialog-command'>
+                    {_("Command")}
+                </label>
+                <input id='run-image-dialog-command'
+                    type='text'
+                    placeholder={_("Command")}
+                    value={dialogValues.command || ''}
+                    onChange={e => this.onValueChanged('command', e.target.value)}
+                    className='form-control' />
+
+                <label className='control-label' htmlFor='run-image-dialog-memory'>
+                    {_("Memory Limit")}
+                </label>
+                <div role='group' className='form-inline'>
+                    <div className="checkbox">
+                        <input id="run-image-dialog-memory-limit-checkbox" type="checkbox"
+                               checked={this.state.memoryConfigure}
+                               onChange={e => this.onValueChanged('memoryConfigure', e.target.checked)} />
+                    </div>
+                    <input className='form-control'
+                           type='number'
+                           value={dialogValues.memory}
+                           step={1}
+                           min={0}
+                           disabled={!this.state.memoryConfigure}
+                           onChange={e => this.onValueChanged('memory', e.target.value)} />
+                    <Select.Select id='memory-unit-select'
+                                   initial={this.state.memoryUnit}
+                                   enabled={this.state.memoryConfigure}
+                                   onChange={value => this.onValueChanged('memoryUnit', value)}>
+                        <Select.SelectEntry data={units.MiB.name} key={units.MiB.name}>
```

Comment:
Any reason why you left out KiB? There's a `unit` for it?

Reply:
Docker supports only MB. I decided to add GB since (at least I) more usually configure systems memory with GB. But enabling the KB select entry is very easy as well. I can add it.

Reply:
Added